src/guide-rngs.md

```diff
@@ -175,40 +175,40 @@ not work well with all algorithms.
 
 The *period* or *cycle length* of a PRNG is the number of values that can be
 generated after which it starts repeating the same random number stream.
-Many PRNGs have a fixed-size period, but for some only an expected average
-cycle length can be given, where the exact length depends on the seed.
-
-On today's hardware, even a fast RNG with a cycle length of *only*
+Many PRNGs have a fixed-size period, while for others ("chaotic RNGs") the
+cycle length may depend on the seed and short cycles may exist.
+
+Note that a long period does not imply high quality (e.g. a counter through
+`u128` values provides a decently long period). Conversely, a short period may
+be a problem, especially when multiple RNGs are used simultaneously.
+In general, we recommend a period of at least 2<sup>128</sup>.
+(Alternatively, a PRNG with shorter period of at least 2<sup>64</sup> and
+support for multiple streams may be sufficient. Note however that in the case
+of PCG, its streams are closely correlated.)
```

Comment:
I think I would not mention streams at all.

Reply:
Since most RNGs providing distinct streams (esp. crypto rngs) already provide sufficient period, agreed.

Reply:
Maybe you should make a difference, because you provide PCG implementations with 128 bits of state:

    /// `Lcg128Xsl64` is also officially known as `pcg64`.
   pub type Pcg64 = Lcg128Xsl64;

In fact, a bit weirdly its source file starts with

    // This is the default multiplier used by PCG for 64-bit state.
   const MULTIPLIER: u128 = 0x2360_ED05_1FC6_5DA4_4385_DF64_9FCC_F645;

which is probably a cut-and-paste error: nobody would use 128-bit constant for 64-bit state.

Reply:
Pcg64 is 128-bit state, 64-bit output (this is how we talk about it and consistent with O'Neil's nominature). Similarly Pcg32 has 64-bit state.

Reply:
OK, keep the claim that you have 64-bit state in the source for a generator with 128-bit state. That will help the reader. A lot. :)

Reply:
Sorry, apparently I can't read!

Reply:
Fixed by https://github.com/rust-random/rand/pull/909